SOX18.py

```diff
@@ -0,0 +1,201 @@
+
+    def __len__(self) -> int:
+        return len(self.seq)
+
+    def __getitem__(self, index: int) -> "Self":
+        return self.__class__(self.seq[index])
+
+    def __str__(self) -> str:
+        return self.seq
+
+    def check_alphabet(self) -> bool:
+        return all([x in self._alphabet for x in self.letters]) and not (
+            "U" in self.letters and "T" in self.letters
+        )
+
+
+class NucleicAcidSequence(ExpressedSequences):
+    def __init__(self, seq: str) -> None:
+        self._alphabet = {"A", "T", "G", "C", "U"}
+        super().__init__(seq, self._alphabet)
+        self._complement_table = None
+
+    def complement(self) -> "Self":
+        if self._complement_table is None:
+            raise NotImplementedError
+        complemented = []
+        for nucleotide in self.seq:
+            complemented.append(self._complement_table[nucleotide])
+        return self.__class__("".join(complemented))
+
+    def gc_content(self) -> int:
+        return (self.seq.count("G") + self.seq.count("C")) / len(self.seq) * 100
+
+
+class DNASequence(NucleicAcidSequence):
+    def __init__(self, seq: str) -> None:
+        super().__init__(seq)
+        self._complement_table = {"T": "A", "A": "T", "C": "G", "G": "C"}
+        self._transcribe_table = {
+            "C": "C",
+            "G": "G",
+            "A": "A",
+            "T": "U",
+        }
+
+    def transcribe(self) -> "RNASequence":
+        transcribed = []
+        for nucleotide in self.seq:
+            transcribed.append(self._transcribe_table[nucleotide])
+        return RNASequence("".join(transcribed))
+
+
+class RNASequence(NucleicAcidSequence):
+    def __init__(self, seq: str):
+        super().__init__(seq)
+        self._complement_table = {"A": "U", "U": "A", "C": "G", "G": "C"}
+
+
+class AminoAcidSequence(ExpressedSequences):
+    def __init__(self, seq: str) -> None:
+        self._alphabet = {
+            "A",
+            "G",
+            "D",
+            "L",
+            "N",
+            "P",
+            "C",
+            "Y",
+            "S",
+            "I",
+            "H",
+            "W",
+            "E",
+            "F",
+            "R",
+            "T",
+            "V",
+            "K",
+            "M",
+            "Q",
+        }
+        super().__init__(seq, self._alphabet)
+        self._positive_aa = {"R", "K", "H"}
+        self._negative_aa = {"D", "E"}
+
+    def define_charge(self) -> Dict[str, int]:
+        positive_count = 0
+        negative_count = 0
+        neutral_count = 0
+        for aa in self.seq:
+            if aa in self._positive_aa:
+                positive_count += 1
+            elif aa in self._negative_aa:
+                negative_count += 1
+            else:
+                neutral_count += 1
+        result = {
+            "Positive": positive_count,
+            "Negative": negative_count,
+            "Neutral": neutral_count,
+        }
+        return result
```

Comment:
Очень круто придумано менять значения в словаре, в дальнейшем, если пускать это в код, наверное, будет очень удобно, но пока это конечный результат функции, который никак потом дальше не используется, для красоты вывода на экран для пользователя, я бы развернула это в строковый вывод какой-нибудь.
```suggestion
        result = {
            "Positive": positive_count,
            "Negative": negative_count,
            "Neutral": neutral_count,
        }
         for key, item in result.items():
             print(f'{key} - {item}')
```